### ISSUE TITLE
Variables in Blade extended anonymous child components have the same values when used more than once

### ISSUE THREAD

**Issue by gavinhewitt** [laravel/framework#32848]
- Laravel Version: 7.11.0
- PHP Version: 7.3
- Database Driver & Version: MySQL 8

### Description:
Trying to wrap a child (anonymous) component with some markup. The inheritance works fine, the child component code is being wrapped correctly. $value however is the same for every item. 

But, if you place $value outside of `@section`, the output is correct. Bug or expected behaviour?

### Steps To Reproduce:

```
<!-- /resources/views/app.blade.php -->

<html>
    <head>
        <title>App Name - @yield('title')</title>
    </head>
    <body>
        <x-child-component value="item-1"/>

        <x-child-component value="item-2"/>
    </body>
</html>


<!-- /resources/views/components/parent.blade.php -->

<p>
    @yield('content')
</p>


<!-- /resources/views/components/child-component.blade.php -->

@extends('components.parent')

@props(['value'])

{{ $value }} <--- print $value outside @section: output correct (item-1, item-2)

@section('content')
    <span>{{ $value }}</span>. <--- output incorrect (item-1, item-1)
@endsection
```

```
// Expected output:

<body>
    <p>
        <span>item-1</span>
    </p>
    <p>
        <span>item-2</span>
    </p>
</body>

// Actual output:

<body>
    <p>
        <span>item-1</span>
    </p>
    <p>
        <span>item-1</span>
    </p>
</body>
```

**Comment by taylorotwell** [comment]
It's very hard to follow your example.

**Comment by gavinhewitt** [comment]
> It's very hard to follow your example. You have a page.blade.php... where is it ever rendered? Share a clearer example please or I'll need to close this. Thanks.

@taylorotwell Sorry, I realise I have not been clear. Example edited.

**Comment by taylorotwell** [comment]
I don't think that's ever going to work because you're re-using the same section name which is not going to function correctly. You should use a component for parent. 

Within `child-component`:

```html
<x-parent-component>
    <span>{{ $value }}</span>.
</x-parent-component>
```

**Comment by gavinhewitt** [comment]
> I don't think that's ever going to work because you're re-using the same section name which is not going to function correctly.

Yeah, that's what I was thinking too. Thanks!
